loader/windows/icd_windows.c

```diff
@@ -103,18 +156,57 @@ BOOL CALLBACK khrIcdOsVendorsEnumerate(PINIT_ONCE InitOnce, PVOID Parameter, PVO
                 KHR_ICD_TRACE("Value not zero, skipping\n");
                 continue;
             }
-
             // add the library
-            khrIcdVendorAdd(cszLibraryName);
+            AdapterAdd(cszLibraryName, ZeroLuid);
         }
+    }
 
-        result = RegCloseKey(platformsKey);
-        if (ERROR_SUCCESS != result)
+    // Add adapters according to DXGI's preference order
+    HMODULE hDXGI = LoadLibrary("dxgi.dll");
+    if (hDXGI)
+    {
+        IDXGIFactory* pFactory = NULL;
+        PFN_CREATE_DXGI_FACTORY pCreateDXGIFactory = (PFN_CREATE_DXGI_FACTORY)GetProcAddress(hDXGI, "CreateDXGIFactory");
```

Comment:
could just use this way instead? https://docs.microsoft.com/en-us/windows/win32/api/dxgi/nf-dxgi-createdxgifactory

Reply:
> 
> 
> could just use this way instead? https://docs.microsoft.com/en-us/windows/win32/api/dxgi/nf-dxgi-createdxgifactory

Saleel,
Are you suggesting to call the API directly without calling GetProcAddress?
The API may not be available on all Windows versions. So, it would be good to go through GetProcAddress.